### PR TITLE
Move confirmDelete from beacon-chain to shared/cmd

### PR DESCRIPTION
**Pull request by 0xKiwi**
This moves the helper function for confirming database deletion to the terminal to avoid unneeded duplication.

This also changes the function to `confirmAction` and allowing the action text and denial text to be passed into the function to make it more general.

### CODE REVIEW

File: shared/cmd/helpers.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+var log = logrus.WithField("prefix", "node")
+
+// ConfirmAction uses the passed in actionText as the confirmation text displayed in the terminal.
+// The user must enter Y or N to indicate whether they confirm the action detailed in the warning text.
+// Returns a boolean representing the user's answer.
+func ConfirmAction(actionText string, deniedText string) (bool, error) {
+	var clearDB bool
+	reader := bufio.NewReader(os.Stdin)
+	log.Warn(actionText)
```
Comment (0xKiwi) [inline shared/cmd/helpers.go]: Should this text be passed into the function and logged there or should we assume it to be logged from outside the function?

File: shared/cmd/helpers.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+	"github.com/sirupsen/logrus"
+)
+
+var log = logrus.WithField("prefix", "node")
+
+// ConfirmAction uses the passed in actionText as the confirmation text displayed in the terminal.
+// The user must enter Y or N to indicate whether they confirm the action detailed in the warning text.
+// Returns a boolean representing the user's answer.
+func ConfirmAction(actionText string, deniedText string) (bool, error) {
+	var clearDB bool
```
Comment (terencechain) [inline shared/cmd/helpers.go]: `clearDB` is not a proper variable name in this context

File: shared/cmd/helpers.go (hunk 1 above)
Comment (terencechain) [reply shared/cmd/helpers.go]: Logging here is good

File: beacon-chain/node/node.go [hunk 3]
```diff
@@ -214,8 +214,22 @@ func (b *BeaconNode) startDB(ctx *cli.Context) error {
 	if err != nil {
 		return err
 	}
-	if clearDB || forceClearDB {
-		d, err = confirmDelete(d, dbPath, forceClearDB)
+	clearDBConfirmed := false
+	if clearDB && !forceClearDB {
+		actionText := "This will delete your beacon chain data base stored in your data directory. " +
```
Comment (nisdas) [inline beacon-chain/node/node.go]: there isnt any need to to pass in the text to the method. The method can log it in there itself

File: shared/cmd/helpers.go [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+		fmt.Print(">> ")
+
+		line, _, err := reader.ReadLine()
+		if err != nil {
+			return false, err
+		}
+		trimmedLine := strings.TrimSpace(string(line))
+		lineInput := strings.ToUpper(trimmedLine)
+		if lineInput != "Y" && lineInput != "N" {
+			log.Errorf("Invalid option of %s chosen, please only enter YzN", line)
```
Comment (nisdas) [inline shared/cmd/helpers.go]:
```suggestion
			log.Errorf("Invalid option of %s chosen, please only enter Y/N", line)
```

File: beacon-chain/node/node.go (hunk 3 above)
Comment (nisdas) [reply beacon-chain/node/node.go]: nvm , it seems like this is a generic function , you can ignore the comment then

File: shared/cmd/helpers.go (hunk 2 above)
Comment (nisdas) [reply shared/cmd/helpers.go]: +1

**Pull request merged**